src/ImageCarouselReact/widget/components/Carousel.tsx

```diff
@@ -0,0 +1,538 @@
+     *
+     * @type {number}
+     * @memberOf ICarouselProps
+     */
+    interval?: number;
+    /**
+     * Show or hide navigation controls
+     *
+     * @type {boolean}
+     * @memberOf ICarouselProps
+     */
+    showControls?: boolean;
+    /**
+     * Pauses the carousel from transitioning when the mouse pointer enters the carousel
+     *
+     * @type {boolean}
+     * @memberOf ICarouselProps
+     */
+    pauseOnHover?: boolean;
+    /**
+     * Specifies whether the carousel should go through all slides continuously, or stop at the last slide
+     *
+     * @type {boolean}
+     * @memberOf ICarouselProps
+     */
+    wrap?: boolean;
+    /**
+     * Callback fired when the active item changes.
+     *
+     * If this callback takes two or more arguments, the second argument will
+     * be a persisted event object with `direction` set to the direction of the
+     * transition.
+     * First argument is the target index i.e Index of the item that triggered it
+     */
+    onSlide?: Function;
+    onSlideEnd?: Function;
+    /**
+     * Index of the current image being showed on the carousel
+     *
+     * @type {number}
+     * @memberOf ICarouselProps
+     */
+    activeIndex?: number;
+    /**
+     * Index of the image that should be shown first
+     *
+     * @type {number}
+     * @memberOf ICarouselProps
+     */
+    defaultActiveIndex?: number;
+    /**
+     * Direction in which the carousel should auto-transition
+     *
+     * @type {Direction}
+     * @memberOf ICarouselProps
+     */
+    direction?: Direction;
+    /**
+     * Icon for the "prev" navigation control
+     *
+     * @type {JSX.Element}
+     * @memberOf ICarouselProps
+     */
+    prevIcon?: JSX.Element;
+    /**
+     * Icon for the "next" navigation control
+     *
+     * @type {JSX.Element}
+     * @memberOf ICarouselProps
+     */
+    nextIcon?: JSX.Element;
+    className?: string;
+    /**
+     * Contains Base properties eg. Base CSS class and prefix for the component
+     * Use only when you want to specify a non-bootstrap properties
+     *
+     * @type {string}
+     * @memberOf ICarouselProps
+     */
+    bsProps?: IBootstrapProps;
+    elementProps?: {};
+};
+
+export type Direction = "prev" | "next";
+
+export interface ICarouselState {
+    activeIndex?: number;
+    previousActiveIndex?: number;
+    direction?: Direction;
+}
+
+export interface ICarouselEvent<T> extends React.MouseEvent<T> {
+    direction: Direction;
+}
+
+class Carousel extends React.Component<ICarouselProps, ICarouselState> {
+    public static defaultProps: ICarouselProps = {
+        bsProps: {
+            bsClass: "carousel",
+        },
+        interval: 5000,
+        nextIcon: <Glyphicon glyph="chevron-right" />,
+        pauseOnHover: true,
+        prevIcon: <Glyphicon glyph="chevron-left" />,
+        showControls: true,
+        showIndicators: true,
+        slide: true,
+        wrap: true,
+    };
+    private isUnmounted: boolean;
```

Comment:
It's best to give them "positive" names if that's their main state. In general the component will be mounted, so `isMounted` is clearer. That's also to avoid double negations (`isUnmounted = false`).